### PR TITLE
Jonmv/vespa cli test runner

### PR DESCRIPTION
**Pull request by jonmv**
@hakonhall please review. Will resolve conflict by reapplying the cli PR before merge.

### CODE REVIEW

File: vespa-osgi-testrunner/src/test/java/com/yahoo/vespa/testrunner/VespaCliTestRunnerTest.java [hunk 1]
```diff
@@ -44,26 +44,27 @@
...
         Path temp = Files.createTempDirectory("vespa-cli-test-runner-test-");
         temp.toFile().deleteOnExit();
-        VespaCliTestRunner runner = new VespaCliTestRunner(temp);
+        Path tests = Files.createDirectory(temp.resolve("tests"));
+        Path artifacts = Files.createDirectory(temp.resolve("artifacts"));
+        VespaCliTestRunner runner = new VespaCliTestRunner(artifacts, tests);
         assertFalse(runner.isSupported());
 
-        Path tests = Files.createDirectory(temp.resolve("tests"));
+        Path systemTests = Files.createDirectory(tests.resolve("system-test"));
```
Comment (hakonhall) [inline vespa-osgi-testrunner/src/test/java/com/yahoo/vespa/testrunner/VespaCliTestRunnerTest.java:52]: This is not used until line 59, so consider moving declaration down.

**Review by hakonhall — APPROVED** [review]

**Comment by jonmv** [comment]
Merging https://github.com/vespa-engine/vespa/pull/20184 should fix conflicts.

File: vespa-osgi-testrunner/src/test/java/com/yahoo/vespa/testrunner/VespaCliTestRunnerTest.java (hunk 1 above)
Comment (jonmv) [reply vespa-osgi-testrunner/src/test/java/com/yahoo/vespa/testrunner/VespaCliTestRunnerTest.java:52]: It's the thing that makes the test "supported", so it needs to stay precisely where it is :)

Comment (hakonhall) [reply vespa-osgi-testrunner/src/test/java/com/yahoo/vespa/testrunner/VespaCliTestRunnerTest.java:52]: Ah, for some reason I didn't see the multiplex declaration (w/ createDirectory()).

**Pull request merged**